### PR TITLE
add OIDC support

### PR DESCRIPTION
**Pull request by ryanlerch**
Update the vagrant setup, andconvert from openid to OIDC

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        container: ["registry.centos.org/centos/centos:7", "registry.fedoraproject.org/fedora:latest", "registry.fedoraproject.org/fedora:rawhide"]
+        container: ["registry.centos.org/centos/centos:7", "registry.fedoraproject.org/fedora:latest"]
```
Comment (adrianreber) [inline .github/workflows/ci.yml:11]: Why is the rawhide test disabled?

Comment (ryanlerch) [reply .github/workflows/ci.yml:11]: whoops. sorry i disabled this and then forgot to investigate further.

it is happening against master too:

https://github.com/fedora-infra/mirrormanager2/runs/4442463827?check_suite_focus=true

not sure how python packaging works with this, but python3-flask is epel only, but python-flask is not. Not sure why it fails in rawhide, but works for latest/f35

 https://src.fedoraproject.org/rpms/python-flask
https://src.fedoraproject.org/rpms/python3-flask

really nore sure how to resolve this

Comment (ryanlerch) [reply .github/workflows/ci.yml]: reenabled the rawhide test

**Comment by ryanlerch** [comment]
I have re-enabled the rawhide test -- so still ready for review.

**Comment by adrianreber** [comment]
Please see https://github.com/fedora-infra/mirrormanager2/pull/303 for a fix. Can you include that commit in your PR? I can also merge #303 and you could rebase.

File: tests/__init__.py [hunk 2]
```diff
@@ -99,7 +102,7 @@ def setUp(self):
         self.session = model.create_tables(DB_PATH, debug=False)
         APP.logger.handlers = []
         APP.logger.setLevel(logging.CRITICAL)
-        APP.before_request(FAS._check_session)
+        #APP.before_request(FAS._check_session)
```
Comment (adrianreber) [inline tests/__init__.py:105]: Why commented out and not removed?

Comment (ryanlerch) [reply tests/__init__.py]: Removed the line now

File: runtests.sh [hunk 3]
```diff
@@ -15,5 +15,5 @@ if [ -x /usr/bin/py.test-3 ]; then
 	# Ensure env uses Python 3
 	mkdir -p .test-bin
 	ln -sf /usr/bin/python3 .test-bin/python
-	PATH="$PWD/.test-bin:$PATH" PYTHONPATH=mirrormanager2 py.test-3 --cov=mirrormanager2 $*
+	PATH="$PWD/.test-bin:$PATH" PYTHONPATH=mirrormanager2 MM2_CONFIG=../tests/mirrormanager2.cfg py.test-3 --cov=mirrormanager2 $*
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply runtests.sh:18]: Added the `MM2_CONFIG=.` part also to the python 2 call above in the same file

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -19,4 +19,4 @@ jobs:
       run: |
         tests/setup_test_container.sh ${{ matrix.container }}
         printenv
-        ./runtests.sh -v
+        MM2_CONFIG=/__w/mirrormanager2/mirrormanager2/tests/mirrormanager2.cfg ./runtests.sh -v
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply .github/workflows/ci.yml]: you're correct. removed this.

File: README.rst [hunk 5]
```diff
@@ -20,11 +20,7 @@ Hacking with Vagrant
 Quickly start hacking on mirrormanager2 using the vagrant setup that is included
 in the repo is super simple.
 
-First, make a copy of the Vagrantfile example::
-
-    $ cp Vagrantfile.example Vagrantfile
-
-Next, install Ansible, Vagrant and the vagrant-libvirt plugin from the official Fedora
+install Ansible, Vagrant and the vagrant-libvirt plugin from the official Fedora
```
[the remark this answers is not in the document]
Comment (ryanlerch) [reply README.rst]: fixed.

**Comment by ryanlerch** [comment]
> Please see #303 for a fix. Can you include that commit in your PR? I can also merge #303 and you could rebase.

Sorry, missed this -- have now rebased against master with your patch merged. all tests are passing now in CI.

cheers.

**Comment by adrianreber** [comment]
Thanks.

**Pull request merged**